backend/lib/endpoints/posts.js

```diff
@@ -40,42 +57,88 @@ async function routes(app) {
       {
         $project: {
           _id: true,
+          authorName: "author.authorName",
+          authorType: "author.authorType",
           commentsCount: {
             $size: "$comments",
           },
-          description: true,
-          likesCount: true,
-          shareWith: true,
-          tags: true,
+          content: true,
+          distance: true,
+          likesCount: {
+            $size: "$likes",
+          },
           title: true,
         },
       },
-    );
-    return Post.aggregate(aggregates);
+    ]);
   });
 
   app.post(
     "/",
-    { preValidation: [app.authenticate], schema: createPostSchema },
-    async (req) => {
-      // todo add logged in user from jwt
-      req.body.authorId = ""; // req.user.id;
-      return new Post(req.body).save();
+    {
+      preValidation: [app.authenticate],
+      schema: createPostSchema,
+    },
+    async (req, reply) => {
+      // TODO: get userId from jwt
+      // userId = ?
+      // user = User.findById(userId);
+      user = await User.findById("5ea6900c0e0419d4cb123611");
+      if (user === null) return new httpErrors.Unauthorized();
+
+      var postProps = req.body;
+
+      // Creates embedded author document
+      postProps.author = {
+        authorId: user.id,
```

Comment:
I think it probably should be `author.name`, because here we have a doc to represent the author's info, regardless of its origin.

For the sync issues, I've put a callback when setting the name at the user model that would handle this, making the data to be always in sync. 
Since the user model is not being migrated yet, we don't have that callback at the user model, but that should handle that and I can cover it in the next PR when changing the User model.

About being either `user.name` or `organization.name`, you mean because both could author posts?  I don't know if I got what you meant.
Here I was going for the current v2 definition with both of them having the same base model, this way we wouldn't have to worry about it here since we could only fetch using the base model.